### PR TITLE
remove deprecated apparmorProfile and redundant type conversion

### PR DESCRIPTION
**Pull request by charles-chenzz**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
remove deprecated apparmor like seccompProfile
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/kubelet/kuberuntime/security_context.go [hunk 1]
```diff
@@ -41,9 +40,6 @@ func (m *kubeGenericRuntimeManager) determineEffectiveSecurityContext(pod *v1.Po
 		return nil, err
 	}
 
-	// set ApparmorProfile.
-	synthesized.ApparmorProfile = apparmor.GetProfileNameFromPodAnnotations(pod.Annotations, container.Name)
```
Comment (saschagrunert) [inline pkg/kubelet/kuberuntime/security_context.go:45]: AppArmor has not graduated yet so I think we still need support for the annotations.

cc @pjbgf

Comment (pjbgf) [reply pkg/kubelet/kuberuntime/security_context.go:45]: Yep, my understanding was that AppArmor annotations would have a similar deprecation strategy as the Seccomp had, in which the annotations were supported for at leat 2 minors after the new field types were introduced. Please refer to the section Version Skew Strategy within the AppArmor graduation KEP: https://github.com/kubernetes/enhancements/pull/3298

Comment (charles-chenzz) [reply pkg/kubelet/kuberuntime/security_context.go:45]: I see. so for now we should wait for the AppArmor to graduate, then we look into this PR, or I close it now and file another one when it's graduate.

**Pull request closed without merging**